lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -1500,12 +1501,39 @@ GfVec3f ProxyRenderDelegate::GetCurveDefaultColor()
 }
 
 //! \brief
-const MColor& ProxyRenderDelegate::GetSelectionHighlightColor(bool lead) const
+MColor ProxyRenderDelegate::GetSelectionHighlightColor(const char* className)
 {
-    static const MColor kLeadColor(0.056f, 1.0f, 0.366f, 1.0f);
+    // Construct the query command string.
+    MString queryCommand;
+    if (className) {
+        queryCommand = "int $index = `displayColor -q -active \"";
+        queryCommand += className;
+        queryCommand += "\"`; colorIndex -q $index;";
+    } else {
+        queryCommand = "displayRGBColor -q \"lead\"";
+    }
+
+    // Query and return the selection color.
+    {
+        MDoubleArray                colorResult;
+        std::lock_guard<std::mutex> _(_mayaCommandEngineMutex);
+        MGlobal::executeCommand(queryCommand, colorResult);
```

Comment:
If a caller passes in a className which isn't valid do we get some kind of error we can warn about?

Reply:
executeCommand runs with displayEnabled = false, so I guess it will not write any error info into log. I think it would be good to set displayEnabled to true here